js/src/Lines.ts

```diff
@@ -592,72 +591,72 @@ export class Lines extends Mark {
         }
     }
 
-    update_dots_xy(animate) {
+    update_dots_xy(animate): void {
         if (this.model.get("marker")) {
-            const x_scale = this.scales.x, y_scale = this.scales.y;
-            const animation_duration = animate === true ? this.parent.model.get("animation_duration") : 0;
+            const xScale = this.scales.x, yScale = this.scales.y;
+            const animationDuration = animate === true ? this.parent.model.get("animationDuration") : 0;
             const dots = this.d3el.selectAll(".curve").selectAll(".dot");
 
-            dots.transition("update_dots_xy").duration(animation_duration)
-                .attr("transform", function(d: any) { return "translate(" + (x_scale.scale(d.x) + x_scale.offset) +
-                        "," + (y_scale.scale(d.y) + y_scale.offset) + ")";
+            dots.transition("update_dots_xy").duration(animationDuration)
+                .attr("transform", function(d: any) { return "translate(" + (xScale.scale(d.x) + xScale.offset) +
+                        "," + (yScale.scale(d.y) + yScale.offset) + ")";
                 })
                 .attr("d", this.dot.size(this.model.get("marker_size"))
                                .type(this.model.get("marker")));
         }
     }
 
-    compute_view_padding() {
+    compute_view_padding(): void {
         //This function sets the padding for the view through the variables
-        //x_padding and y_padding which are view specific paddings in pixel
-        let x_padding;
+        //xPadding and yPadding which are view specific paddings in pixel
+        let xPadding;
         if (this.model.get("marker")) {
-            const marker_padding = Math.sqrt(this.model.get("marker_size")) / 2 + 1.0;
-            const line_padding = this.model.get("stroke_width") / 2.0;
-            x_padding = Math.max(marker_padding, line_padding);
+            const markerPadding = Math.sqrt(this.model.get("marker_size")) / 2 + 1.0;
+            const linePadding = this.model.get("stroke_width") / 2.0;
+            xPadding = Math.max(markerPadding, linePadding);
         } else {
-            x_padding = this.model.get("stroke_width") / 2.0;
+            xPadding = this.model.get("stroke_width") / 2.0;
         }
 
-        const y_padding = x_padding;
-        if(x_padding !== this.x_padding || y_padding !== this.y_padding) {
-            this.x_padding = x_padding;
-            this.y_padding = y_padding;
+        const yPadding = xPadding;
+        if(xPadding !== this.xPadding || yPadding !== this.yPadding) {
+            this.xPadding = xPadding;
+            this.yPadding = yPadding;
             this.trigger("mark_padding_updated");
         }
     }
 
     update_marker_style() {
         const that = this;
```

Comment:
Maybe get rid of that one

Reply:
Hum, so this is a difficult one. As you can see

```
    update_marker_style() {
        const that = this;
        const fillColor = this.model.get("fill_colors");
        const opacities = this.model.get("opacities");
        this.d3el.selectAll(".curve").each(function(d, i) {
            const curve = d3.select(this);
            curve.selectAll(".dot")
                .style("opacity", opacities[i])
                .style("fill:", that.get_mark_color(d, i) || fillColor[i]);
        });
    }
```

We're making use of both objects here, both the `this` from inside the scope of the function the `this` of the outside scope ;/
I looked a bit into `d3` docs and I believe that's the recommended way of doing it, with `each` and using `select.this`, I tried to be more specific and get the element but I couldn't, it's just data... not a DOM object, meaning it doesn't have info like `parent` and stuff.
I don't really know how to address to this issue other than using `that`, please let me know if you have an idea!

Reply:
Indeed. That's an annoying one. Let's keep it that way for now then :)

Reply:
hehe yeah, but if you have input on how to fix it I'm interested!

Reply:
I remember we had to deal with something similar before but I don't recall where.